modules/geoipbackend/geoipbackend.cc

```diff
@@ -629,6 +637,20 @@ string GeoIPBackend::format2str(string sformat, const Netmask& addr, GeoIPNetmas
       rep = queryGeoIP(addr, GeoIPInterface::Name, tmp_gl);
     } else if (!sformat.compare(cur,3,"%ci")) {
       rep = queryGeoIP(addr, GeoIPInterface::City, tmp_gl);
+    } else if (!sformat.compare(cur,4,"%nsp")) {
+      rep = queryGeoIP(addr, GeoIPInterface::ISP, tmp_gl);
+      for (long unsigned int i = 0; i < rep.length(); i++) {
```

Comment:
This for-loop would be more efficient using the right boost replace function, like you did in other places.